src/sentry/spans/consumers/process/flusher.py

```diff
@@ -26,6 +27,74 @@
 logger = logging.getLogger(__name__)
 
 
+class MultiProducerManager:
+    """
+    Manages multiple Kafka producers for load balancing across brokers/topics.
+
+    Configure multiple producers in settings.py:
+
+    KAFKA_MULTI_PRODUCER_CONFIGS = {
+        "buffered-segments": [
+            {"cluster": "default", "topic": "buffered-segments-1"},
+            {"cluster": "secondary", "topic": "buffered-segments-2"}
+        ]
+    }
+    """
+
+    def __init__(self, topic: Topic):
+        self.topic = topic
+        self.producers: list[KafkaProducer] = []
+        self.topics: list[ArroyoTopic] = []
+        self.current_index = 0
+        self._setup_producers()
+
+    def _setup_producers(self):
+        """Setup producers based on configuration."""
+        # Get multi-producer config if available, otherwise use single producer
+        multi_config = getattr(settings, "KAFKA_MULTI_PRODUCER_CONFIGS", {}).get(self.topic.value)
+
+        if multi_config:
+            # Multiple producers configured
+            for config in multi_config:
+                cluster_name = config["cluster"]
+                topic_name = config.get(
+                    "topic", get_topic_definition(self.topic)["real_topic_name"]
```

Comment:
When would we rely on the default here ? In other words, is it expected that `topic` may not exist in `config` ?